### ISSUE TITLE
[PPC] upc_resetphase() fails - argument mismatch

### ISSUE THREAD

**Issue by nenadv** [Intrepid/clang-upc#45]
I rebuilt PPC version and noticed the following failure.  Here is the simplified version of the test:
```
include <stdio.h>
#include <upc.h>
#include <gula.h>

shared [5] int a[5*THREADS];
int
main()
{
    shared [5] int *p = &a[3];
    shared [5] int *pr;

    pr = upc_resetphase(p); /* should reset phase */

    if (upc_phaseof(p) != 3 || upc_phaseof(pr) != 0)
        GULA_FAIL("failed in upc_phaseof() check - 2");

    upc_barrier;

    if (!MYTHREAD)
        printf ("Passed.\n");
}
```
When `upc_resetphase()` is called, register R4 is set to the UPC pointer value, or the value is being placed on the stack:
```
12          pr = upc_resetphase(p); /* should reset phase */
(gdb) x/10i $pc
=> 0x10001c18 <main+264>:       std     r3,256(r31)
   0x10001c1c <main+268>:       std     r3,56(r1)
   0x10001c20 <main+272>:       ld      r4,256(r31)
   0x10001c24 <main+276>:       addi    r3,r31,264
   0x10001c28 <main+280>:       bl      0x100021ac <.upc_resetphase>
```
but `upc_resetphase()` expects pointer in R3:
```
=> 0x100021ac <.upc_resetphase>:        std     r3,-16(r1)
   0x100021b0 <.upc_resetphase+4>:      ld      r3,-16(r1)
   0x100021b4 <.upc_resetphase+8>:      std     r3,-24(r1)
   0x100021b8 <.upc_resetphase+12>:     ld      r3,-24(r1)
   0x100021bc <.upc_resetphase+16>:     lis     r4,-16
   0x100021c0 <.upc_resetphase+20>:     and     r3,r3,r4
   0x100021c4 <.upc_resetphase+24>:     std     r3,-24(r1)
   0x100021c8 <.upc_resetphase+28>:     ld      r3,-24(r1)
   0x100021cc <.upc_resetphase+32>:     blr
```
I recall that in GUPC we had to change PPC ABI for something like this.

**Comment by nenadv** [comment]
After merge to 3.5/3.6 it seems that the above fix does not work for struct pointer representation.  Intrepid's test18 fails on upc_resetphase() with similar symptoms. Looking into llvm IR code:

- upc_resetphase (C code) receives two 64 bit values, and return two 64 bit values
- UPC caller to upc_resetphase allocates temporary for the return and passes pointer to it

I tried to remove indirect return, but then something else broke.

**Comment by nenadv** [comment]
ABI for PPC has been changed to v2.  By default they return 16 bytes aggregates in two registers (dorect).  Removed the previous patch.

Fixed by this commit 4f500bf26a9175643c6567b71a37259cb9996dbf
